src/button/ReactionToggleButton.test.tsx

```diff
@@ -0,0 +1,253 @@
+    <TestComponent rtcSession={rtcSession} room={room} />,
+  );
+  act(() => {
+    getByRole("button").click();
+  });
+  act(() => {
+    getByRole("button", {
+      name: "Search",
+    }).click();
+  });
+  await act(async () => {
+    getByPlaceholderText("Search reactions…").focus();
+    await user.keyboard("crickets");
+  });
+  expect(container).toMatchSnapshot();
+  act(() => {
+    getByText("🦗").click();
+  });
+  expect(room.testSentEvents).toEqual([
+    [
+      undefined,
+      ElementCallReactionEventType,
+      {
+        "m.relates_to": {
+          event_id: memberEventAlice,
+          rel_type: "m.reference",
+        },
+        name: "crickets",
+        emoji: "🦗",
+      },
+    ],
+  ]);
+});
+
+test("Can search for and send emoji with the keyboard", async () => {
+  const user = userEvent.setup();
+  const room = new MockRoom(memberUserIdAlice);
+  const rtcSession = new MockRTCSession(room, membership);
+  const { getByRole, getByPlaceholderText, container } = render(
+    <TestComponent rtcSession={rtcSession} room={room} />,
+  );
+  act(() => {
+    getByRole("button").click();
+  });
+  act(() => {
+    getByRole("button", {
+      name: "Search",
+    }).click();
+  });
+  const searchField = getByPlaceholderText("Search reactions…");
+  await act(async () => {
+    searchField.focus();
+    await user.keyboard("crickets");
+  });
+  expect(container).toMatchSnapshot();
+  act(() => {
+    fireEvent.keyDown(searchField, { key: "Enter" });
+  });
+  expect(room.testSentEvents).toEqual([
+    [
+      undefined,
+      ElementCallReactionEventType,
+      {
+        "m.relates_to": {
+          event_id: memberEventAlice,
+          rel_type: "m.reference",
+        },
+        name: "crickets",
+        emoji: "🦗",
+      },
+    ],
+  ]);
+});
+
+test("Can close search", () => {
+  const room = new MockRoom(memberUserIdAlice);
+  const rtcSession = new MockRTCSession(room, membership);
+  const { getByRole, container } = render(
+    <TestComponent rtcSession={rtcSession} room={room} />,
+  );
+  act(() => {
+    getByRole("button").click();
+  });
+  act(() => {
+    getByRole("button", {
+      name: "Search",
+    }).click();
+  });
+  act(() => {
+    getByRole("button", {
+      name: "close search",
+    }).click();
+  });
+  expect(container).toMatchSnapshot();
+});
+
+test("Can close search with the escape key", () => {
+  const room = new MockRoom(memberUserIdAlice);
+  const rtcSession = new MockRTCSession(room, membership);
+  const { getByRole, container, getByPlaceholderText } = render(
+    <TestComponent rtcSession={rtcSession} room={room} />,
+  );
+  act(() => {
+    getByRole("button").click();
+  });
+  act(() => {
+    getByRole("button", {
+      name: "Search",
+    }).click();
+  });
+  const searchField = getByPlaceholderText("Search reactions…");
+  act(() => {
+    fireEvent.keyDown(searchField, { key: "Escape" });
```

Comment:
This should also have a user-event equivalent

Reply:
I couldn't find a way to fire a control character from user events, so had to resort to this. This seemed to be what Stack Overflow suggests :disappointed:

Reply:
I believe it looks like: `await user.keyboard("[Escape]");`